### PR TITLE
Update the AdaDelta optimizer class API docs 📗

### PR DESCRIPTION
**Pull request by 8bitmp3**
Hi all. Updating the AdaDelta optimizer class API docs here:

- Description: expanded, based on Zeiler's 2012 original paper, Keras and TF 2.x docs.
- New parameters section:
    - (Note: TF 2.x AdaDelta API has different default LR and epsilon values defined in the class in `/optimizer_v2/` compared with another `optimizer.py`, the research paper, and Keras' implementation. It's not a new issue but I raised it just in case: https://github.com/tensorflow/tensorflow/issues/36785.)
- Reference: updated (author, year of publishing).
- New example code snippet: it's an autoencoder adapted from /swift-models/ with comments. It was tested in Colab on FashionMNIST separately:
    - The 9th epoch input/output looked like this, if anyone is curious:
        input: image output: image
- To keep things consistent with the new params section, I expanded the definition of `ρ` in the code comments (the words are from the original paper.) Hope you don't mind that.

Working on the rest of the API docs.

Enjoy your weekend.

### CODE REVIEW

**Comment by dan-zheng** [comment]
All Swift code in [tensorflow/swift-apis](https://github.com/tensorflow/swift-apis) was automatically formatted via [`swift-format`](https://github.com/apple/swift-format) in https://github.com/tensorflow/swift-apis/pull/711.

[Formatting configuration here](https://github.com/tensorflow/swift-apis/blob/master/.swift-format).
One notable change is that we're now using 2-space indentation, which is consistent with:
- [Google Swift Style Guide](https://google.github.io/swift)
- The Swift standard library style
- The style of [apple/swift-format](https://github.com/apple/swift-format)

---

As a result, this pull request now has merge conflicts due to code formatting changes. Please resolve the conflicts if you'd like us to re-review your patch! If you haven't already, please consider installing [`swift-format`](https://github.com/apple/swift-format) so you can auto-format code by yourself in the future!

File: Sources/TensorFlow/Optimizers/MomentumBased.swift [hunk 1]
```diff
@@ -105,20 +105,69 @@ public class AdaGrad<Model: Differentiable>: Optimizer
...
+/// ````
 public class AdaDelta<Model: Differentiable>: Optimizer
     where Model.TangentVector: VectorProtocol & PointwiseMultiplicative & ElementaryFunctions,
           Model.TangentVector.VectorSpaceScalar == Float {
     public typealias Model = Model
     /// The learning rate.
     public var learningRate: Float
-    /// The decay factor, corresponding to fraction of gradient to keep at each time step.
+    /// The decay factor of the previous parameter update/a fraction of gradient
+    /// to keep at each time step.
```
Comment (dan-zheng) [inline Sources/TensorFlow/Optimizers/MomentumBased.swift:170]: I wonder what's the rationale behind the documentation change for `var rho`?
The current doc comment matches https://keras.io/optimizers/#adadelta:
> **rho**: float >= 0. Adadelta decay factor, corresponding to fraction of gradient to keep at each time step.

Comment (dan-zheng) [reply Sources/TensorFlow/Optimizers/MomentumBased.swift]: I reverted this change temporarily, since the rationale isn't clear to me.
